src/website/shared/listeners/cache_suggestions.py

```diff
@@ -295,3 +301,32 @@ def parse_drv_name(name: str) -> tuple[str, str]:
         return match.group(1), match.group(2)
     else:
         return name, ""
+
+
+def maintainers_list(packages: dict, edits: list[MaintainersEdit]) -> list[dict]:
+    """
+    Returns a deduplicated list (by GitHub ID) of all the maintainers of all the
+    affected packages linked to this suggestion, modified by potential
+    user-supplied edits.
+    """
+
+    to_remove = {
+        m.maintainer for m in edits if m.type == MaintainersEdit.EditType.REMOVE
+    }
+    to_add = [m.maintainer for m in edits if m.type == MaintainersEdit.EditType.ADD]
```

Comment:
Why is this a list and the other thing a set?

Reply:
We use them differently: `to_remove` is basically a hashset that we query to know if we should add a maintainer to the working list. I realize, by the way, that it shoud be only github ids and not the whole maintainers. Maybe a better name is `to_skip` and it should be merged with `seen` below; they have the same role.

`to_add` is a list of maintainer to process and add at the end, but we deduplicate already when adding (and we need to do it by github_id, and not on the whole maintainer object as I mentioned below in another comment). We could make `to_add` a set but it's useless. Deduplication happen later.